Automattic/wp-calypso
# Add desktop-development environment

With [wp-desktop](https://github.com/Automattic/wp-desktop) we are moving away from the makefile setup and moving to npm scripts to harmonise wp-desktop and wp-calypso. It should become easier to build and deploy new releases and developers should be able to easily test their changes directly in the app without cumbersome manual rebuilds.

**Follow the main PR here:** https://github.com/Automattic/wp-desktop/pull/440

## Context:
To run calypso in wp-desktop in development mode, including live reloading and all the amazing little helpers that make life as a developer significantly easier, we need a new calypso environment called  `desktop-development`. (`desktop` is the production environment)

## Setup & Test:
### Setup:
In WP-Desktop, checkout this PR: https://github.com/Automattic/wp-desktop/pull/446

Then initialize calypso and checkout this PR:
```
git submodule init
git submodule update

cd calypso
git checkout -b add/desktop-development-env origin/add/desktop-development-env
```
In the wp-calypso folder, run the calypso and desktop dev server with:
```
make dev-server
```

As soon as the calypso dev server is up and running, start the app from another terminal session with:
```
make dev
```

### Test:
- Go to the Dev Tools console (<kbd>Cmd</kbd>+<kbd>Shift</kbd>+<kbd>i</kbd>) and check if calypso is being served from `calypso.localhost:3000` e.g. via `window.location.href`
- Login to your account using the oauth login (`/oauth-login`)
